packages/mom5/package.py

```diff
@@ -0,0 +1,317 @@
+# .f, .f90, .F, .F90. Given a sourcefile <file>.<ext>, where <ext> is one of
+# the above, this provides a number of default actions:
+
+# make <file>.T		create a CIF file
+# make <file>.lst	create a compiler listing
+# make <file>.o		create an object file
+# make <file>.s		create an assembly listing
+# make <file>.x		create an executable file, assuming standalone
+#			source
+
+# make <file>.i		create a preprocessed file (only for .F and .F90
+#			extensions)
+
+# make <file>.hpm	produce hpm output from <file>.x
+# make <file>.proc	produce procstat output from <file>.x
+
+# The macro TMPFILES is provided to slate files like the above for removal.
+
+RM = rm -f
+SHELL = /bin/csh
+TMPFILES = .*.m *.T *.TT *.hpm *.i *.lst *.proc *.s
+
+.SUFFIXES: .F .F90 .H .T .f .F90 .h .hpm .i .lst .proc .o .s .x
+
+.f.T:
+	$(FC) $(FFLAGS) -c -Ca $*.f
+.f.lst:
+	$(FC) $(FFLAGS) $(LIST) -c $*.f
+.f.o:
+	$(FC) $(FFLAGS) -c     $*.f
+.f.s:
+	$(FC) $(FFLAGS) -eS    $*.f
+.f.x:
+	$(FC) $(FFLAGS) $(LDFLAGS) -o $*.x $*.f
+.f90.T:
+	$(FC) $(FFLAGS) -c -Ca $*.f90
+.f90.lst:
+	$(FC) $(FFLAGS) $(LIST) -c $*.f90
+.f90.o:
+	$(FC) $(FFLAGS) -c     $*.f90
+.f90.s:
+	$(FC) $(FFLAGS) -c -eS $*.f90
+.f90.x:
+	$(FC) $(FFLAGS) $(LDFLAGS) -o $*.x $*.f90
+.F.T:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) -c -Ca $*.F
+.F.i:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) -eP $*.F
+.F.lst:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) $(LIST) -c $*.F
+.F.o:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) -c     $*.F
+.F.s:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) -c -eS $*.F
+.F.x:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) $(LDFLAGS) -o $*.x $*.F
+.F90.T:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) -c -Ca $*.F90
+.F90.i:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) -eP $*.F90
+.F90.lst:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) $(LIST) -c $*.F90
+.F90.o:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) -c     $*.F90
+.F90.s:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) -c -eS $*.F90
+.F90.x:
+	$(FC) $(CPPDEFS) $(CPPFLAGS) $(FFLAGS) $(LDFLAGS) -o $*.x $*.F90
+.x.proc:
+	procstat -R $*.proc $*.x
+.x.hpm:
+	hpm -r -o $*.hpm $*.x
+"""
+
+        with open(makeinc_path, "w") as makeinc:
+            makeinc.write(config[self.compiler.name] + config["post"])
+
+    def build(self, spec, prefix):
+
+        # cd ${ACCESS_OM_DIR}/src/mom/exp
+        # export mom_type=ACCESS-OM
+        # ./MOM_compile.csh --type $mom_type --platform spack
+
+        # cppDefs = "-Duse_netCDF -Duse_netCDF3 -Duse_libMPI -DUSE_OCEAN_BGC -DENABLE_ODA -DSPMD -DLAND_BND_TRACERS"
+        # set executable    = $root/exec/$platform/$type/fms_$type.x
+        # set srcList = ( accessom_coupler )
+        # set includes = "-I$executable:h:h/lib_FMS -I$executable:h:h/$type/lib_ocean"
+        # set libs = "$executable:h:h/$type/lib_ocean/lib_ocean.a $executable:h:h/lib_FMS/lib_FMS.a"
+        # bin/mkmf -f -m Makefile -a src -t bin/mkmfs.template.spack -p $executable:t" -o "$includes" -c "$cppDefs" -l "$libs"  $srcList
```

Comment:
```suggestion
        # Build replicates the following shell command
        '''
        cd ${ACCESS_OM_DIR}/src/mom/exp
        export mom_type=ACCESS-OM
        ./MOM_compile.csh --type $mom_type --platform spack
        '''
```
I don't think the snippet from the `MOM_compile.csh` script is worth including.

Reply:
OK, happy to remove it. Please note that if we want to avoid using the `MOM_compile.csh`, that is what we'd have to do from this file.

Reply:
Yes I understood the intent of including it, but I thought it was confusing and a little misleading because that is just one bit of the file, so for example:
```
cppDefs = "-Duse_netCDF -Duse_netCDF3 -Duse_libMPI -DUSE_OCEAN_BGC -DENABLE_ODA -DSPMD -DLAND_BND_TRACERS"
```
is set as the default at the beginning (https://github.com/ACCESS-NRI/MOM5/blob/master/exp/MOM_compile.csh#L80) but later it is modified (https://github.com/ACCESS-NRI/MOM5/blob/master/exp/MOM_compile.csh#L93):
```
    set cppDefs  = ( "-Duse_netCDF -Duse_libMPI -DACCESS_OM" )
```
and again (https://github.com/ACCESS-NRI/MOM5/blob/master/exp/MOM_compile.csh#L117)
```
    set cppDefs = "$cppDefs -Duse_netCDF4"
```

Now I'm not arguing this is good design, but including the snippet just isn't that useful because of the convoluted logic in the build script.